### PR TITLE
cleanup enc_ctx_size

### PR DESCRIPTION
**Pull request by danielsn**
*Description of changes:*
As part of the CBMC proof effort, cleanup ```aws_cryptosdk_enc_ctx_size```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by danielsn** [comment]
This update uses some new features from C-Common, and will fail until the c-common version is updated.

File: source/enc_ctx.c [hunk 1]
```diff
@@ -31,34 +31,39 @@ int aws_cryptosdk_enc_ctx_init(struct aws_allocator *alloc, struct aws_hash_tabl
...
+    AWS_PRECONDITION(AWS_OBJECT_PTR_IS_WRITABLE(size));
+    AWS_PRECONDITION(aws_hash_table_is_valid(enc_ctx));
 
+    const size_t entry_count = aws_hash_table_get_entry_count(enc_ctx);
+    if (entry_count > UINT16_MAX) return aws_raise_error(AWS_CRYPTOSDK_ERR_LIMIT_EXCEEDED);
+    if (entry_count == 0) {
+        // Empty context.
+        *size = 0;
+        AWS_POSTCONDITION(aws_hash_table_is_valid(enc_ctx));
+        AWS_POSTCONDITION(*size <= UINT16_MAX);
```
Comment (ttjsu-aws) [inline source/enc_ctx.c]: We just set  *size =0 in the prev statements, what is this purpose of this post condition?

File: source/enc_ctx.c [hunk 2]
```diff
@@ -31,34 +31,39 @@ int aws_cryptosdk_enc_ctx_init(struct aws_allocator *alloc, struct aws_hash_tabl
 }
 
 int aws_cryptosdk_enc_ctx_size(size_t *size, const struct aws_hash_table *enc_ctx) {
-    size_t serialized_len = 2;  // First two bytes are the number of k-v pairs
-    size_t entry_count    = 0;
+    AWS_PRECONDITION(AWS_OBJECT_PTR_IS_WRITABLE(size));
+    AWS_PRECONDITION(aws_hash_table_is_valid(enc_ctx));
 
+    const size_t entry_count = aws_hash_table_get_entry_count(enc_ctx);
+    if (entry_count > UINT16_MAX) return aws_raise_error(AWS_CRYPTOSDK_ERR_LIMIT_EXCEEDED);
```
Comment (ttjsu-aws) [inline source/enc_ctx.c]: We use this styling for our if statements in the code. 
```
if () {
...
}
```

**Comment by danielsn** [comment]
Superceded by https://github.com/aws/aws-encryption-sdk-c/pull/423

**Pull request closed without merging**